src/bitmessagekivy/baseclass/addressbook.py

```diff
@@ -0,0 +1,171 @@
+            exist_addresses = len(self.ids.ml.children)
+            if exist_addresses != len(self.queryreturn):
+                self.update_addressBook_on_scroll(exist_addresses)
+            self.has_refreshed = (
+                True if exist_addresses != len(self.queryreturn) else False
+            )
+
+    def update_addressBook_on_scroll(self, exist_addresses):
+        """Load more data on scroll down"""
+        self.set_mdList(exist_addresses, exist_addresses + 5)
+
+    @staticmethod
+    def refreshs(*args):
+        """Refresh the Widget"""
+
+    # @staticmethod
+    def addBook_detail(self, address, label, instance, *args):
+        """Addressbook details"""
+        if instance.state == 'closed':
+            instance.ids.delete_msg.disabled = True
+            if instance.open_progress == 0.0:
+                obj = AddbookDetailPopup()
+                self.address_label = obj.address_label = label
+                self.address = obj.address = address
+                width = .9 if platform == 'android' else .8
+                self.addbook_popup = self.address_detail_popup(
+                    self.send_message_to, self.update_addbook_label, self.close_pop,
+                    width=width, obj=obj)
+                self.addbook_popup.auto_dismiss = False
+                self.addbook_popup.open()
+        else:
+            instance.ids.delete_msg.disabled = False
+
+    def delete_address(self, address, instance, *args):
+        """Delete inbox mail from inbox listing"""
+        self.ids.ml.remove_widget(instance.parent.parent)
+        # if len(self.ids.ml.children) == 0:
+        if self.ids.ml.children is not None:
+            self.ids.tag_label.text = ''
+        sqlExecute(
+            "DELETE FROM  addressbook WHERE address = '{}';".format(address))
+        toast('Address Deleted')
+
+    def close_pop(self, instance):
+        """Pop is Canceled"""
+        self.addbook_popup.dismiss()
+        toast('Canceled')
+
+    def update_addbook_label(self, instance):
+        """Updating the label of address book address"""
+        address_list = kivy_helper_search.search_sql(folder="addressbook")
+        stored_labels = [labels[0] for labels in address_list]
+        add_dict = dict(address_list)
+        label = str(self.addbook_popup.content_cls.ids.add_label.text)
+        if label in stored_labels and self.address == add_dict[label]:
+            stored_labels.remove(label)
+        if label and label not in stored_labels:
+            sqlExecute(
+                "UPDATE addressbook SET label = '{}' WHERE"
+                " address = '{}';".format(
+                    label, self.addbook_popup.content_cls.address))
+            state.kivyapp.root.ids.sc11.ids.ml.clear_widgets()
+            state.kivyapp.root.ids.sc11.loadAddresslist(None, 'All', '')
+            self.addbook_popup.dismiss()
+            toast('Saved')
+
+    def send_message_to(self, instance):
```

Comment:
there should be a UI-independent function for doing this, it can use for example dependency injection. E.g.

```
def compose_message(compose_screen, from=None, to=None, subject=None, body=None):
    if from:
        populate_from
    if to
        populate_to
    if subject:
        populate_subject
    if message:
        populate_message
```